nuxeo-drive-jsf/src/main/java/org/nuxeo/drive/seam/NuxeoDriveActions.java

```diff
@@ -167,10 +173,15 @@ public boolean canEditDocument(DocumentModel doc) {
         return getFileSystemItem(doc) != null;
     }
 
-    public boolean hasOneDriveToken(Principal user) {
-        TokenAuthenticationService tokenService = Framework.getLocalService(TokenAuthenticationService.class);
+    public boolean hasOneDriveToken(Principal user) throws UnsupportedEncodingException {
+        TokenAuthenticationService tokenService = Framework.getService(TokenAuthenticationService.class);
         for (DocumentModel token : tokenService.getTokenBindings(user.getName())) {
-            if ("Nuxeo Drive".equals(token.getPropertyValue("authtoken:applicationName"))) {
+            String applicationName = (String) token.getPropertyValue("authtoken:applicationName");
+            if (applicationName == null) {
+                continue;
+            }
+            if (NUXEO_DRIVE_APPLICATION_NAME.equals(
```

Comment:
Could you mention in a comment (if I'm not mistaken) that we do the url decoding for "backward compat" reasons, but that in the future tokens should be stored in their natural format (i.e. not needing re-decoding).